llmfoundry/data/finetuning/dataloader.py

```diff
@@ -145,6 +148,14 @@ def build_finetuning_dataloader(cfg: DictConfig, tokenizer: Tokenizer,
         )
 
     else:
+        if '://' in cfg.dataset.hf_name:
```

Comment:
recommend using `parse_uri` (also from `from composer.utils import parse_uri`)

Reply:
but this could also be valid with a value like `sam-mosaic/dolly-hhrlhf`

Reply:
i don't follow, what could also be valid? It looks like this code path is only supposed to execute when the `cfg.dataset.hf_name` is a `{s3|gcp|oci}://` uri path.

Reply:
Are you saying something like:

```python
if '://' in cfg.dataset.hf_name:
  backend, bucket_name, path = parse_uri(cfg.dataset.hf_name)
```

or are you saying that there is a way to use `parse_uri` to not do the `'://' in` check?

Reply:
oh, maybe

```python
backend, bucket_name, path = parse_uri(cfg.dataset.hf_name)
if backend != '':
```
?

Reply:
yeah, the second thing you said. except double check if its backend `None` or empty string